File: make-payment/make-payment.go
package main

import (
	"fmt"
	"github.com/iotaledger/iota.go/account/builder"
	"github.com/iotaledger/iota.go/account/deposit"
	"github.com/iotaledger/iota.go/account/oracle"
	oracle_time "github.com/iotaledger/iota.go/account/oracle/time"
	"github.com/iotaledger/iota.go/account/store/badger"
	"github.com/iotaledger/iota.go/account/timesrc"
	"github.com/iotaledger/iota.go/api"
	"time"
)

// You should never hard-code a seed
var seed = "PUETTSEITFEVEWCWBTSIZM9NKRGJEIMXTULBACGFRQK9IMGICLBKW9TTEVSDQMGWKBXPVCBMMCXWMNPDX"

func main() {
	// Define the node to connect to
	apiSettings := api.HTTPClientSettings{URI: "https://nodes.devnet.iota.org:443"}
	iotaAPI, err := api.ComposeAPI(apiSettings)
	handleErr(err)

	nodeInfo, err := iotaAPI.GetNodeInfo()
	handleErr(err)
	fmt.Println("Connected to a Devnet node: " + nodeInfo.AppName)

	store, err := badger.NewBadgerStore("db")
	handleErr(err)

	// Make sure the database closes when the code stops
	defer store.Close()

	// create an accurate time source (in this case Google's NTP server).
	timesource := timesrc.NewNTPTimeSource("time.google.com")

	account, err := builder.NewBuilder().
		// Connect to the node
		WithAPI(iotaAPI).
		// Create the database
		WithStore(store).
		WithSeed(seed).
		// Set the minimum weight magnitude for the Devnet
		WithMWM(9).
		WithTimeSource(timesource).
		// Load the default plugins that enhance the functionality of the account
		WithDefaultPlugins().
		Build()
	handleErr(err)

	handleErr(account.Start())

	// Make sure the account shuts down when the code stops
	defer account.Shutdown()

	// Define the CDA to send the payment to
	magnetLink := "iota://BWNYWGULIIAVRYOOFWZTSDFXFPRCFF9YEHGVBOORLGCPCJSKTHU9OKESUGZGWZXZZDLESFPPTGEHVKTTXG9BQLSIGP/?timeout_at=5174418337&multi_use=1&expected_amount=0"

	// Parse the magnet link
	cda, err := deposit.ParseMagnetLink(magnetLink)
	handleErr(err)

	// Set the oracle's threshold to 30 minutes
	threshold := time.Duration(30) * time.Minute
	// Create the oracle
	timeDecider := oracle_time.NewTimeDecider(timesource, threshold)
	// Create a n instance of the oracle
	sendOracle := oracle.New(timeDecider)

	// Ask the oracle if the CDA is OK to send to
	ok, rejectionInfo, err := sendOracle.OkToSend(cda)
	handleErr(err)
	if !ok {
		fmt.Println("Won't send transaction: ", rejectionInfo)
		return
	}

	// Create and send the bundle
	bundle, err := account.Send(cda.AsTransfer())
	handleErr(err)

	fmt.Printf("Sent deposit to %s in the bundle with the following tail transaction hash %s\n", cda.Address, bundle[0].Hash)

}

func handleErr(err error) {
	if err != nil {
		panic(err)
	}
}

File: java/account-module/src/main/java/com/iota/MakePayment.java
package com.iota;

import org.iota.jota.IotaAPI;
import org.iota.jota.IotaAccount;
import org.iota.jota.account.AccountStore;
import org.iota.jota.account.deposits.ConditionalDepositAddress;
import org.iota.jota.account.deposits.methods.DepositFactory;
import org.iota.jota.account.deposits.methods.MagnetMethod;
import org.iota.jota.account.store.AccountFileStore;
import org.iota.jota.error.ArgumentException;
import org.iota.jota.model.Bundle;
import java.io.File;
import java.util.Optional;
import java.util.concurrent.ExecutionException;

class MakePayment {
    public static void main(String[] args) throws ArgumentException, InterruptedException, ExecutionException {

    // Connect to a node
    IotaAPI api = new IotaAPI.Builder()
        .protocol("https")
        .host("nodes.devnet.thetangle.org")
        .port(443)
        .build();

    // The seed that the account uses to generate CDAs and send bundles
    String mySeed = "PUEOTSEITFEVEWCWBTSIZM9NKRGJEIMXTULBACGFRQK9IMGICLBKW9TTEVSDQMGWKBXPVCBMMCXWMNPDX";
    
    // Create a file to store the seed state
    File file = new File("seed-state-database.json");
    AccountStore store = new AccountFileStore(file);
    
    // Create an account
    IotaAccount account = new IotaAccount.Builder(mySeed)
    .store(store)
    .api(api)
    .build();

    // Start the account and any plugins
    account.start();

    // Define the CDA to send the payment to
    String magnet = "iota://LQHETLMIBUEGAHMHMIQFBOMSSF9BF9FHOFWAFGDLND9CMBDXOPLBDPMXHBEGZSFFDYVHSXEXCBYVDXKQYXOWVOCRYB/?timeout_at=1575452386589&multi_use=true&expected_amount=0";

    ConditionalDepositAddress cda = DepositFactory.get().parse(magnet, MagnetMethod.class);

    // Send the bundle
    Bundle bundle = account.send(
        cda.getDepositAddress().getHashCheckSum(), 
        cda.getRequest().getExpectedAmount(), 
        Optional.of("Thanks for the pizza"),
        Optional.of("ACCOUNTMODULETEST")).get();

    System.out.printf("Sent deposit to %s in the bundle with the following tail transaction hash %s\n",
    bundle.getTransactions().get(bundle.getLength() - 1).getAddress(), bundle.getTransactions().get(bundle.getLength() - 1).getHash());
    
    // Close the database and stop any ongoing reattachments
    account.shutdown();
    }
}

